src/exec_context.ts

```diff
@@ -198,11 +215,29 @@ export default class ExecContext<M> {
       return fn.call(this, ...args);
     };
 
+    const mergeContainerEnv = (parent?: ExecContext<M>, env?: Environment): Environment => {
+      if (!env && (!parent || !parent.env)) {
+        return null;
+      }
+
+      if (!env) {
+        return parent.env;
+      }
+
+      if (!parent || !parent.env) {
+        return env;
+      }
+
+      const mergeEffects = (k, l, r) => k === 'effects' ? mergeMap(l, r) : r;
+      return environment(mergeDeepWithKey(mergeEffects, parent.env.identity(), env.identity()));
+    };
+
+    const contaierEnv = mergeContainerEnv(parent, env);
     const wrapInit = (props: string[]) => pipe(pick(props), map(pipe(fn => fn.bind(this), initialize)));
-    const errLog = error(env.log);
+    const errLog = contaierEnv && error(contaierEnv.log) || (() => { });
```

Comment:
What do you `errLog` to default to with no container?

Reply:
Same as now? https://github.com/ai-labs-team/casium/pull/16/files#diff-f98485da1849ae631ba5c04e5cec07ddR141

Reply:
Also `contaier` looks like a typo.